docs/DesignDocs/Attestation_API_Proposal.md

```diff
@@ -685,6 +685,130 @@ since legacy applications don't call these API functions explicitly.
+
+| Format ID | Evidence structure  |
+| -- | - |
+| 1. `OE_FORMAT_UUID_SGX_ECDSA_P256` | `oe_report_header \|\| SGX_ECDSA_quote(hash) \|\| custom_claims` |
+| 2. `OE_FORMAT_UUID_SGX_ECDSA_P256 + flag` | `oe_attestation_header \|\| oe_report_header \|\| SGX_ECDSA_quote(hash) \|\| custom_claims` |
+| 3. `OE_FORMAT_UUID_SGX_ECDSA_P256_REPORT` | `oe_report_header \|\| SGX_ECDSA_quote(custom_claims)` |
+| 4. `OE_FORMAT_UUID_SGX_ECDSA_P256_REPORT + flag` | `oe_attestation_header \|\| oe_report_header \|\| SGX_ECDSA_quote(custom_claims)` |
+| 5. `OE_FORMAT_UUID_SGX_ECDSA_P256_QUOTE` | `SGX_ECDSA_quote(custom_claims)` |
+| 6. `OE_FORMAT_UUID_SGX_ECDSA_P256_QUOTE + flag` | `oe_attestation_header \|\| SGX_ECDSA_quote(custom_claims)` |
+
+For all evidence structures above, the hearts of the evidences are the same - SGX ECDSA_P256 quote. SGX ECDSA_P256 quote should to be like any other evidence format which is presented in 2 formats - with or without embedded format id. For example, format XYZ is presented as:
+1. Without flag - `Evidence_XYZ`
+2. With flag - `oe_attestation_header || Evidence_XYZ || custom_claims`
+
+That is, we only need to support these evidence structures:
+- Format 2: Generated by previous version oe_get_evidence().
+- Format 3: Generated by legacy oe_get_report() which will eventually be deprecated.
+- Format 5: Generated by SGX SDK.
+
+In the proposal, flag `OE_EVIDENCE_FLAGS_EMBED_FORMAT_ID` indicates wraping the plugin evidence with a header containing format ID, and hashing the custom claims and passing the hash to the plugin.
+
+Consider renaming the flag to reflect the fact.
+
+### User Experience
+
+#### Attester
+- To generate format 2:
+```
+oe_get_evidence(
+    OE_FORMAT_UUID_SGX_ECDSA_P256,
+    OE_EVIDENCE_FLAGS_EMBED_FORMAT_ID,
+    custom_claims,
+    size_t custom_claims_size,
+    ...);
+```
+
+- To generate format 3:
+```
+oe_get_evidence(
+    OE_FORMAT_UUID_SGX_ECDSA_P256_REPORT,
+    0,  // Must be 0
+    custom_claims,
+    size_t custom_claims_size,  // Must be <= 64 bytes
+    ...);
+```
+
+- To generate format 5:
+```
+oe_get_evidence(
+    OE_FORMAT_UUID_SGX_ECDSA_P256,
+    0,
+    custom_claims, 
+    size_t custom_claims_size,  // Must be <= 64 bytes
+    ...);
+```
+
+#### Verifier
+- To verify format 2:
+```
+oe_verify_evidence(
+    NULL,
+    ...);
+```
+
+- To verify format 3:
+```
+oe_verify_evidence(
+    OE_FORMAT_UUID_SGX_ECDSA_P256_REPORT,
+    ...);
+```
+
+- To verify format 5:
+```
+oe_verify_evidence(
+    OE_FORMAT_UUID_SGX_ECDSA_P256,
+    ...);
+```
+
+### Plan
+Reduction/deprecation is planned in phases.
+
+Here 'Remove' means the API definition is removed from public header files. Referring to the removed API definition results in compilation error.
+'Deprecate' means the API definition is redefined in public header files. Referring to the deprecated API definition does not break the build but the behavior is changed internally to guarantee continuous support.
```

Comment:
Can you elaborate on what it means for the API to be redefined in the public header files? It's a little different from my standard understanding of what it means for an API to be deprecated, which is that:
1. The API is marked as deprecated. This means that documentation (e.g. doxygen, change logs) assert that the use of the API is not recommended, and tools may start to warn if it used.
2. The API signature remains unchanged and behaves as it did previously. (i.e. it is still a breaking change and not deprecation if an API alters existing behavior).

Reply:
Examples of 'redefine'
#define oe_get_report(x,y,z) oe_get_evidence(SGX_UUID, x, y, z, NULL, ...)

Code that call oe_get_report() still compiles but it's replaced by new API. Call stack is different but the outcome remain the same.

Reply:
I think that description would be useful to include in the line items that talk about the deprecate `oe_get_report` and `oe_verify_report` functions, rather than have it be indirectly referenced in an atypical definition of "deprecate" (e.g. I imagine that it doesn't apply in the same way to deprecation of `OE_FORMAT_UUID_SGX_ECDSA_P256_REPORT`)